climada/engine/impact.py

```diff
@@ -1372,7 +1372,147 @@ def _selected_exposures_idx(self, coord_exp):
+           raise ValueError(f"Concatenation option {concat_type} not available.")
+       
+        #initialize new impact object
+        impact=cls()
+       
+        # check impacts consistency and fill impact object
+       
+        ### the following checks are common for both concat options
+       
+        # check hazard type
+        haz_types = {imp.tag['haz'].haz_type for imp in imp_list if imp.tag['haz'].haz_type != ''}
+        if len(haz_types) > 1:
+           raise ValueError(f"The impacts are based on hazards of different types: {haz_types}. "
+                            "The impacts are incompatible and cannot be concatenated.")      
+       
+        # check unit
+        units = {imp.unit for imp in imp_list if imp.unit != ''}
+        if len(units) > 1:
+           raise ValueError(f"The given impacts have different units: {units}. "
+                            "The impacts are incompatible and cannot be concatenated.")
+        if len(units) == 0:
+           units = {''}
+        impact.unit = units.pop()
+
+        #check coordinate reference system
+        crss={imp.crs for imp in imp_list if imp.crs != ''}
+        if len(crss)>1:
+           raise ValueError(f"The given impacts have different crs: {crss}. "
+                        "The impacts are incompatible and cannot be concatenated.")
+        impact.crs=crss.pop()
+ 
+        #check attributes
+        attributes = sorted(set.union(*[set(vars(imp).keys()) for imp in imp_list]))
+        for attr_name in attributes:
+           if not all(hasattr(imp, attr_name) for imp in imp_list):
+               raise ValueError(f"Attribute {attr_name} is not shared by all impacts. "
+                                "The impacts are incompatible and cannot be concatenated.")
+
+        #check if impacts have same frequency values (only first entries are checked)
+        freqs=set([imp.frequency[0] for imp in imp_list])
+        if len(freqs)>1:
+           raise ValueError(f"Impacts do not have the same frequency: {freqs}."
+                            "The impacts are incompatible and cannot be concatenated.")
+        frequencies=[imp.frequency for imp in imp_list]
+        impact.frequency = np.concatenate(frequencies,axis=0) 
+       
+        ### The following checks are specific to the selected concatenation options
+       
+        if concat_type=='time_window':
+          
+          #check if no overlapping dates
+          dates=[date for imp in imp_list for date in imp.date]
+          
+          if len(dates)!=len(set(dates)): 
+              raise ValueError("There is at least one date in more than one impact. Please make sure impacts do not contain same dates."
+                               "The impacts are incompatible and cannot be concatenated.")
+          impact.date = np.array(dates)
+          
+          
+          #check if exposure is consistent
+          isin=[np.isin(imp.coord_exp,imp_list[0].coord_exp) for imp in imp_list[0::]]
```

Comment:
This will fail. You cannot check 2-d arrays with `np.isin`, as it will consider each number as independent. What you really want to do here is to compare coordinates, so pair of numbers. 

If you want to check that they are equal, use `numpy.array_equal`.

Reply:
agreed, I changed it using numpy.array_equal